File: LeetCode/java/1062_Longest_Repeating_Substring.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int longestRepeatingSubstring(String s) {
        int answer = 0;
        Map<String, Integer> map = new HashMap<>();

        for (int i = s.length() - 1; i >= 2; i--) {
            map.clear();

            StringBuilder temp = new StringBuilder();

            for (int j = 0; j < i; j++) {
                temp.append(s.charAt(j));
            }

            map.put(temp.toString(), map.getOrDefault(temp.toString(), 0) + 1);

            temp.deleteCharAt(0);

            for (int j = i; j < s.length(); j++) {
                temp.append(s.charAt(j));

                map.put(temp.toString(), map.getOrDefault(temp.toString(), 0) + 1);
                temp.deleteCharAt(0);
            }

            for (Map.Entry<String, Integer> m : map.entrySet()) {
                if (m.getValue() > 1) {
                    if (answer < m.getKey().length()) {
                        answer = m.getKey().length();
                    }
                }
            }

            if (answer > 0) {
                break;
            }
        }

        return answer;
    }
}

File: LeetCode/cpp/1062_Longest_Repeating_Substring.cpp
class Solution {
public:
	int longestRepeatingSubstring(string s) {
		int answer = 0;
		unordered_map<string, int> um;

		for (int i = s.length() - 1; i >= 2; i--) {
			um.clear();

			string temp;

			for (int j = 0; j < i; j++) {
				temp += s[j];
			}

			um[temp]++;

			temp.erase(0, 1);

			for (int j = i; j < s.length(); j++) {
				temp += s[j];

				um[temp]++;
				temp.erase(0, 1);
			}

			for (auto m : um) {
				if (m.second > 1) {
					if (answer < m.first.length()) {
						answer = m.first.length();
					}
				}
			}

			if (answer > 0) {
				break;
			}
		}

		return answer;
	}
};